### PR TITLE
update for julia 0.6

### PR DESCRIPTION
**Pull request by bjarthur**
@tlnagy this is needed for Gadfly

### CODE REVIEW

**Comment by bjarthur** [comment]
tests are failing for julia 0.4.  do we just want to drop support?

**Comment by yuyichao** [comment]
Or use `@compat` in the test.

File: test/runtests.jl [hunk 1]
```diff
@@ -5,7 +5,7 @@ using Base.Test
 
 srand(100)
 xs = 10 .* rand(100)
-ys = sin(xs) .+ 0.5 * rand(100)
+ys = sin.(xs) .+ 0.5 * rand(100)
```
Comment (tlnagy) [inline test/runtests.jl]: You probably need a `@compat` here

**Comment by bjarthur** [comment]
i did a rebase and screwed up the commits.  will close this PR and try afresh

**Pull request closed without merging**